corehq/apps/app_manager/models.py

```diff
@@ -352,7 +362,28 @@ def get_property_names(self):
 
     @property
     def is_subcase(self):
-        return self.parent_tag
+        return bool(self.parents)
+
+    @classmethod
+    def wrap(cls, data):
+        _assert = soft_assert(to=['nhooper@dimagi.com'])
+        if not _assert('parent_tag' not in data,
+                       'AdvancedAction not migrated. Old application build may have been reverted.'):
+            if data['parent_tag']:
+                # Lazy-migrate single parent index to list of parent/host indices
+                parent = {
+                    'tag': data['parent_tag'],
+                    'reference_id': data.get('parent_reference_id', 'parent'),
+                    'relationship': data.get('relationship', 'child')
+                }
+                if hasattr(data.get('parents'), 'append'):
+                    data['parents'].append(parent)
+                else:
+                    data['parents'] = [parent]
+                del data['parent_tag']
+                data.pop('parent_reference_id', None)
```

Comment:
We can use `del` because we know the key exists. If `parent_reference_id` isn't in `data`, `del` will throw a KeyError. `somedict.pop(somekey, None)` won't, but it will do the same thing as `del`. It's more wordy, so I like `del` more.